test/common/upstream/eds_test.cc

```diff
@@ -1152,6 +1152,214 @@ TEST_F(EdsTest, PriorityAndLocalityWeighted) {
   EXPECT_EQ(1UL, stats_.counter("cluster.name.update_no_rebuild").value());
 }
 
+TEST_F(EdsTest, EndpointUpdateHealthCheckConfig) {
```

Comment:
Arguably, you could subclass the test fixture `EdsTest` and add in a bunch of helper methods to make these tests less verbose; there is a lot of boilerplate that could be refactored away to make the tests more readable that way.